sdk/src/main/java/com/kin/ecosystem/history/presenter/OrderHistoryPresenter.java

```diff
@@ -53,6 +59,8 @@ public void onAttach(IOrderHistoryView view) {
 		eventLogger.send(OrderHistoryPageViewed.create());
 		getOrderHistoryList();
 		listenToCompletedOrders();
+		// Trigger to update balance, if someone paid to this user(p2p).
+		blockchainSource.getBalance(null);
```

Comment:
the balance presenter should handle this logic. When ever the balance presser is attached it should add observer to balance updated and remove it when it detached.